tests/cross/src/test/scala/tests/tokens/SemanticTokensSuite.scala

```diff
@@ -292,4 +292,33 @@ class SemanticTokensSuite extends BaseSemanticTokensSuite {
         |}
         |""".stripMargin,
   )
+
+  // When for-comprehension includes line with `=`, we get `scala.x$1`, `scala.x$2` symbols on `foo`.
+  // Both `scala` and `x$#` have position on `foo`, and we don't want to highlight it as a `scala` package,
+  // so we need `namespace` to have lower priority than `variable`.
+  check(
+    "for-comprehension",
+    s"""|package <<example>>/*namespace*/
+        |
+        |object <<B>>/*class*/ {
+        |  val <<a>>/*variable,definition,readonly*/ = for {
+        |    <<foo>>/*parameter,declaration,readonly*/ <- <<List>>/*class*/("a", "b", "c")
+        |    _ = <<println>>/*method*/("print!")
+        |  } yield <<foo>>/*variable,readonly*/
+        |}
+        |""".stripMargin,
+    compat = Map(
+      "3" ->
+        """|package <<example>>/*namespace*/
+           |
+           |object <<B>>/*class*/ {
+           |  val <<a>>/*variable,definition,readonly*/ = for {
+           |    <<foo>>/*variable,definition,readonly*/ <- <<List>>/*class*/("a", "b", "c")
+           |    <<_>>/*class,abstract*/ = <<println>>/*method*/("print!")
```

Comment:
When LSH is `_`, we get `$1$` symbol, but with incorrect span and symbol of result type of RHS (in this case, `Unit`) with span on `_`, so this `class,abstract` is `Unit`. This happens only in tests, since `SemanticTokensProvider` always makes `_` a `variable,readonly`. I can maybe change the test and mention in there, so it won't be confusing